distributed/tests/test_worker.py

```diff
@@ -407,6 +407,29 @@ def __str__(self):
         assert "Bar" in str(e.__cause__)
 
 
+@pytest.mark.asyncio
+async def test_plugin_exception():
+    class MyException(Exception):
+        def __init__(self, msg):
+            self.msg = msg
+
+        def __str__(self):
+            return "MyException(%s)" % self.msg
+
+    class MyPlugin:
+        def setup(self, worker=None):
+            raise MyException("Foo")
```

Comment:
Could we use a built-in exception instead, for example `raise ValueError("setup failed")`? Then later ensure that the expected error was raised with `pytest.raises(ValueError, match="setup failed")`